vm/src/handlers/lib/ecall.rs

```diff
@@ -844,10 +852,13 @@ impl<'a, const N: usize> CommEcallHandler<'a, N> {
                 .read_buffer(path.0, &mut path_local_raw[0..(path_len * 4)])?;
         }
 
-        // convert to a slice of u32, by taking 4 bytes at the time as big-endian integers
-        let path_local = unsafe {
-            core::slice::from_raw_parts(path_local_raw.as_ptr() as *const u32, path_len as usize)
-        };
+        // read bytes and combine into u32 values safely (avoid unaligned access)
+        let mut path_local = Vec::with_capacity(path_len as usize);
+        for i in 0..path_len {
+            let idx = (i * 4) as usize;
+            let bytes = &path_local_raw[idx..idx + 4];
+            path_local.push(u32::from_be_bytes([bytes[0], bytes[1], bytes[2], bytes[3]]));
```

Comment:
As it turns out, the ARM Cortex processors we use in Ledger devices use little endian (docs were indeed incorrect), so

```suggestion
            path_local.push(u32::from_le_bytes([bytes[0], bytes[1], bytes[2], bytes[3]]));
```
should fix the integration tests.

I might consider changing the specs, but that's better done in a separate PR anyway.